crates/bdk/src/descriptor/error.rs

```diff
@@ -10,7 +10,7 @@
 // licenses.
 
 //! Descriptor errors
-
+#[cfg(feature = "std")]
```

Comment:
What is the impact for `no-std`?
What would be the returned errors in `no-std`?

Reply:
I config'd out `fmt::Display` for errors here for consistence since it was done in other parts of the bdk package. Technically I think we could leave it and it wouldn't break `no_std` code but I believe it would make the final binary larger which could be a problem for some `no_std` embedded devices.  

@afilini any thoughts on if we should or should not include `fmt::Display` impls for errors with `no_std`?

Reply:
I don't think it will hurt to impl Display for `no_std`. Also curious on the reasoning.

Reply:
To answer @realeinherjar there would still be returned error enums but just no formatted .to_string(). 

I also found this list of suggestions for making small `no_std` binaries, including the recommendation to not use `core::fmt`. Looks like it only applies if someone is using `bdk` and needs an executable smaller than 20 kilobytes.

[Remove core::fmt with #![no_main] and Careful Usage of libstd](https://github.com/johnthagen/min-sized-rust#remove-corefmt-with-no_main-and-careful-usage-of-libstd)

Reply:
Since this change would have to be done for all bdk crates for it to have any effect on `no_std` binary size (and I don't think are blocked by this) I created a new issue #1198 targeted for the 2.0 milestone.